packages/myst-cli/src/process/file.ts

```diff
@@ -108,9 +111,84 @@ export function loadTexFile(
   return { kind: SourceFileKind.Article, mdast: tex.ast as GenericParent, frontmatter };
 }
 
-export function loadMySTJSON(content: string) {
-  const { mdast, frontmatter, kind } = JSON.parse(content);
-  return { kind: kind ?? SourceFileKind.Article, mdast, frontmatter };
+export function mystJSONValidationOpts(
+  vfile: VFile,
+  opts?: { property?: string; ruleId?: RuleId },
+): ValidationOptions {
+  return {
+    property: opts?.property ?? 'file',
+    file: vfile.path,
+    messages: {},
+    errorLogFn: (message: string) => {
+      fileError(vfile, message, { ruleId: opts?.ruleId ?? RuleId.mystJsonValid });
+    },
+    warningLogFn: (message: string) => {
+      fileWarn(vfile, message, { ruleId: opts?.ruleId ?? RuleId.mystJsonValid });
+    },
+  };
+}
+
+function validateMySTJSON(
+  input: any,
+  opts: ValidationOptions,
+): { mdast: GenericParent; kind: SourceFileKind; frontmatter: PageFrontmatter } {
+  const value = validateObjectKeys(
+    input,
+    { required: ['mdast'], optional: ['kind', 'frontmatter'] },
+    opts,
+  );
+  // Although this is required, we'll set it to an empty article by default
+  const mdast: GenericParent = value?.mdast ?? { type: 'root', children: [] };
```

Comment:
Is this a reasonable thing to do? Declare this `mdast` property `required`, but add a default? I did this so that we don't need to throw an error in this function.